lib/watir_drops/page_object.rb

```diff
@@ -97,8 +115,27 @@ def fill_form(model)
     def inspect
       '#<%s url=%s title=%s>' % [self.class, url.inspect, title.inspect]
     end
+
     alias selector_string inspect
 
+
+    def on_page?
+      if self.respond_to?('page_url')
+        Watir::Wait.until { page_url.gsub(/.*:\/\//i,'').gsub(/\/$/i,'') == (@browser.url.gsub(/.*:\/\//i,'').gsub(/\/$/i,'')) }
+      end
+
+      if self.respond_to?('page_title')
+        Watir::Wait.until { @browser.title == page_title }
+      end
+
+      if self.class.required_element_list.any?
+        Watir::Wait.until { self.class.required_element_list.all? { |e| send(e).present? } }
+      end
+
```

Comment:
we might want to rescue timeouts and throw more specific errors, but I'm not sure how descriptive the default error is because there are no tests. ;-)